ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/ConnectIntegrationTest.java

```diff
@@ -239,7 +254,70 @@ public void shouldReturnError() {
         containsString("Connector mock-connector already exists"));
   }
 
+  @Test
+  public void shouldReadTimestampsFromConnect() {
+    // Given:
+    create("mock-source", ImmutableMap.<String, String> builder()
+        .put("connector.class", "org.apache.kafka.connect.tools.VerifiableSourceConnector")
+        .put("topic", "foo")
+        .put("throughput", "5")
+        .put("id", "123")
+        .put("topic.creation.default.replication.factor", "1")
+        .put("topic.creation.default.partitions", "1")
+        .build());
+
+    final long start = System.nanoTime();
+    RestResponse<KsqlEntityList> response;
+    do {
+      response = ksqlRestClient.makeKsqlRequest("CREATE STREAM FOO (PAYLOAD TIMESTAMP) WITH (KAFKA_TOPIC='foo', VALUE_FORMAT='DELIMITED');");
+    } while(!response.isSuccessful() && System.nanoTime() - start < TIMEOUT_NS);
+
+    // When:
+    final RestResponse<List<StreamedRow>> queryFoo = ksqlRestClient.makeQueryRequest("SELECT * FROM FOO EMIT CHANGES LIMIT 1;", 1L);
+
+    // Then:
+    assertThat("successfully queried FOO", queryFoo.isSuccessful());
+    assertThat(queryFoo.getResponse().get(1).getRow().get().getColumns().get(0), is("1970-01-01T00:00:00.000"));
+  }
+
+  @Test
+  @Ignore
```

Comment:
Setting this to ignore, because reading from stdout from a separate library seems to not work with Jenkins (and is probably not a good idea either). Swapping out `org.apache.kafka.connect.tools.VerifiableSinkConnector` with a connector to a real database and checking the values written to the database would work, but it would be a lot more effort than it is worth.